pkg/executor/core/executor.go

```diff
@@ -121,13 +115,7 @@ func (executor Executor) verifySubjectInternal(ctx context.Context, verifyParame
 		return types.VerifyResult{}, ReferrersNotFound
 	}
 
-	overallVerifySuccess := true
-	for _, referenceType := range verifyParameters.ReferenceTypes {
-		if anyVerifySuccess[referenceType] == false {
-			overallVerifySuccess = false
-			break
-		}
-	}
+	overallVerifySuccess := executor.PolicyEnforcer.OverallVerifyResult(ctx, verifierReports)
```

Comment:
So we don't short circuit the execution if any of the verification result fails when the policy is set to all right?

Reply:
Yes you're right. I missed this. I suggest that we make the `ContinueVerifyOnFailure` a more generic method called `ContinueVerify` which can determine both in the success and failure case of the current verifier report if it should proceed. That way it can short circuit on first failure for 'all' policy and first success for 'any'. Does it make sense?

Reply:
We currently don't short circuit on success for the 'any' policy for that particular artifact type. There is redundant checking that occurs for that particular artifact type if there's already been a success. I think to fix this we would need to keep track of which artifact types we've seen and evaluated to true already...